cosmwasm/enclaves/shared/contract-engine/src/contract_validation.rs

```diff
@@ -675,24 +1151,167 @@ fn verify_message_params(
     }
     let msg = msg.unwrap();
 
-    if msg.sender() != Some(&signer_addr) {
-        warn!(
-            "message signer did not match cosmwasm message sender: {:?} {:?}",
-            signer_addr, msg
-        );
-        return false;
+    info!("Verifying message sender...");
+    match msg {
+        CosmosSdkMsg::MsgRecvPacket { .. }
+        | CosmosSdkMsg::MsgAcknowledgement { .. }
+        | CosmosSdkMsg::MsgTimeout { .. } => {
+            // No sender to verify.
+            // Going to pass null sender to the contract if all other checks pass.
+        }
```

Comment:
Why not just use a default case for this if there's no logic anyway?

Reply:
I liked it better when it's explicit 🤷‍♂️

Reply:
LMK what you think. When it's explicit there's less room for errors later on.

For example, if we add a new message type in the future and it falls under the default case, the sender won't be checked, and the compiler won't alert us to add logic here for that type.